### PR TITLE
feat: conditional-area

### PR DESCRIPTION
**Pull request by Eerie-Cson**
Conditional area using object as parameter/s

### CODE REVIEW

File: basic/conditional-area.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  constructor (shape){
+    this.shape = shape;
+  }
+  area(arg)
+  {
+    
+    if(this.shape == 'circle'){
+      if(Object.keys(arg).length>1){
+        throw new Error("Wrong parameters for the given shape: Circle");
+      }
```
Comment (ralphcasipe1) [inline basic/conditional-area.js]: You checked if the available `args` are greater than one.
If I tried to pass an argument like this:
```js
shape.area({width: 10 })
```
This will pass.

The best way to assert is to check if there is a key named `radius` in an object and only have one key. Otherwise, throw an error.

File: basic/conditional-area.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  it('should throw error, given the shape pentagon', function(){
+    const shape = new Shape('pentagon');   
+    expect(()=> shape.area({base: 2, height: 10})).toThrow("Shape is not available");
+  });
+
+  it('should throw error, given the wrong parameter', function(){
+    const shape = new Shape('circle');  
+    expect(()=>shape.area({width: 10, height:10 })).toThrow("Wrong parameters for the given shape: Circle");
+  });
```
Comment (ralphcasipe1) [inline basic/conditional-area.test.js]: Try adding another unit tests:
```js
const errorMessage = 'Wrong parameters for the given shape: Circle';
expect(()=>shape.area({})).toThrow(errorMessage);
expect(()=>shape.area({width: 10 })).toThrow(errorMessage);
expect(()=>shape.area({radius: 10, width: 10 })).toThrow(errorMessage);
```

File: basic/conditional-area.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+describe('ConditionalArea', function(){
+  
+  it('should calculate circle area', function(){
+    const shape = new Shape('circle');  
+    expect(shape.area({radius: 10})).toEqual(314.1592653589793);
+  });
+
+  it('should calculate rectangle area', function(){
+    const shape = new Shape('rectangle');   
+    expect(shape.area({width: 10, height: 5})).toEqual(50);
```
Comment (ralphcasipe1) [inline basic/conditional-area.test.js:17]: Try adding a test that has an empty object, `{}` and expect an error

File: basic/conditional-area.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  it('should calculate triangle area', function(){
+   const shape = new Shape('triangle');   
+    expect(shape.area({base: 2, height: 10})).toEqual(10);
```
Comment (ralphcasipe1) [inline basic/conditional-area.test.js:23]: Try adding additional keys in the object and empty object, expect an error on it.

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]

File: basic/conditional-area.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  it('should throw error, given the shape pentagon', function(){
+    const shape = new Shape('pentagon');   
+    expect(()=> shape.area({base: 2, height: 10})).toThrow("Shape is not available");
+  });
+/*
+  it('should throw error, given the wrong parameter', function(){
+    const shape = new Shape('circle');  
+    ;
+  });
+*/
```
Comment (ralphcasipe1) [inline basic/conditional-area.test.js]: Remove commented code

**Review by ralphcasipe1 — CHANGES_REQUESTED** [review]

File: basic/conditional-area.js (hunk 1 above)
Comment (ralphcasipe1) [reply basic/conditional-area.js]: **RESOLVED** 💯

File: basic/conditional-area.test.js (hunk 2 above)
Comment (ralphcasipe1) [reply basic/conditional-area.test.js]: **RESOLVED** 💯

**Review by ralphcasipe1 — APPROVED** [review]

File: basic/conditional-area.test.js (hunk 3 above)
Comment (ralphcasipe1) [reply basic/conditional-area.test.js:17]: Resolved 💯

File: basic/conditional-area.test.js (hunk 4 above)
Comment (ralphcasipe1) [reply basic/conditional-area.test.js:23]: Resolved 💯
